protocol-units/da/movement/protocol/da/src/mock/mod.rs

```diff
@@ -64,7 +64,7 @@ where
 
 impl<C> DaOperations<C> for Mock<C>
 where
-	C: Curve + Send + Sync + 'static,
+	C: Curve + Send + Sync + 'static + std::fmt::Debug,
```

Comment:
Why was this bound added here, with no new code to require it?

Reply:
Yes it has been removed because block debug for testing has been removed but as it can be useful to be able to debug the Da I keep it.

Reply:
Well, it does not do anything in this particular impl, so it may be better to remove it unless it's forced by the other bounds on `C` that were added elsewhere.

I suspect it's just some lazy `#[derive(Debug)]` that requires the phantom curve parameter to have `Debug` as well, even though it's not needed.

Reply:
All the debug bound I've added has been requested by the compiler when I add some debut code to understand the DA issues.
I keep them because it can be useful to sort DA issue.